core/src/vaults.ts

```diff
@@ -0,0 +1,446 @@
+    const incentiveConstantDaiHex = await liquidatorContract.tip();
+    const incentiveRelativeDaiHex = await liquidatorContract.chip();
+    const incentiveRelativeDai = new BigNumber(incentiveRelativeDaiHex._hex)
+        .shiftedBy(-WAD_NUMBER_OF_DIGITS)
+        .multipliedBy(debtDai);
+    const incentiveConstantDai = new BigNumber(incentiveConstantDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    const incentiveCombinedDai = incentiveRelativeDai.plus(incentiveConstantDai);
+    return {
+        incentiveCombinedDai,
+        incentiveConstantDai,
+        incentiveRelativeDai,
+    };
+};
+
+export const fetchVaultLiquidationIncentive = memoizee(_fetchVaultLiquidationIncentive, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 3,
+});
+
+const _fetchLiquidatedParameters = async (network: string, vault: Vault) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const typeHex = ethers.utils.formatBytes32String(vault.collateralType);
+    const eventFilter = contract.filters.Bark(typeHex, vault.address, null, null, null, null, null);
+    const liquidationEvents = await contract.queryFilter(eventFilter);
+    if (liquidationEvents.length !== 0 && vault.initialDebtDai.eq(0)) {
+        // there was a liquidation and the vault was not used again
+        return Promise.all(
+            liquidationEvents.map(async event => ({
+                liquidationDate: await fetchDateByBlockNumber(network, event.blockNumber),
+                transactionHash: event.transactionHash,
+                auctionId: `${vault.collateralType}:${new BigNumber(event.args?.id._hex).toFixed(0)}`,
+            }))
+        );
+    }
+    return undefined;
+};
+export const fetchLiquidatedParameters = memoizee(_fetchLiquidatedParameters, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const getAuctionedDaiAndAuctionState = (proximityToLiquidation: BigNumber, vault: Vault, debtDai: BigNumber) => {
+    // logic from https://github.com/makerdao/dss/blob/fa4f6630afb0624d04a003e920b0d71a00331d98/src/dog.sol#L186
+    // detemines if the vault is liquidatable and what amount of debt can be covered.
+    let state: 'liquidatable' | 'not-liquidatable' = proximityToLiquidation.isLessThanOrEqualTo(0)
+        ? 'liquidatable'
+        : 'not-liquidatable';
+    const amountDaiCanBeAuctionedGloballyDai = vault.maximumProtocolDebtDai.minus(vault.currentProtocolDebtDai);
+    const amountDaiCanBeAuctionedCollateralDai = vault.maximumCollateralDebtDai.minus(vault.currentCollateralDebtDai);
+    const minimumDebtCovered = amountDaiCanBeAuctionedCollateralDai.isLessThan(amountDaiCanBeAuctionedGloballyDai)
+        ? amountDaiCanBeAuctionedCollateralDai
+        : amountDaiCanBeAuctionedGloballyDai;
+    if (amountDaiCanBeAuctionedGloballyDai.eq(0) || amountDaiCanBeAuctionedCollateralDai.eq(0)) {
+        state = 'not-liquidatable';
+    }
```

Comment:
Why exactly 0 is a separate edge case? Don't we need to check if it's smaller OR equal than that?

Reply:
From the contract i'vd read that this difference cannot fall below 0. But comparing with <= instead of === is safer ;)